spec/services/audit/replication_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Audit::Replication do
+  let(:results) { described_class.results(preserved_object) }
+
+  let(:preserved_object) { create(:preserved_object, current_version: 2) }
+  let(:endpoints) { preserved_object.zipped_moab_versions.map(&:zip_endpoint).uniq }
+  let(:endpoint) { endpoints.first }
+  let(:endpoint2) { endpoints.second }
+  let(:zipped_moab_version1) { preserved_object.zipped_moab_versions.where(zip_endpoint: endpoint).first }
+  let(:zipped_moab_version2) { preserved_object.zipped_moab_versions.where(zip_endpoint: endpoint).second }
+  let(:zipped_moab_version3) { preserved_object.zipped_moab_versions.where(zip_endpoint: endpoint2).first }
+  let(:zipped_moab_version4) { preserved_object.zipped_moab_versions.where(zip_endpoint: endpoint2).second }
+  let(:audit_class1) { endpoint.audit_class }
+  let(:audit_class2) { endpoint2.audit_class }
+
+  before do
+    # allow(Audit::ReplicationSupport).to receive(:logger).and_return(logger)
+    # creation of MoabRecord triggers archive zip creation, as archive zips are created from moabs
+    create(:moab_record, preserved_object: preserved_object, version: preserved_object.current_version)
+    allow(Audit::ReplicationSupport).to receive(:check_child_zip_part_attributes).and_return(false)
+    allow(audit_class1).to receive(:check_replicated_zipped_moab_version)
+    allow(audit_class2).to receive(:check_replicated_zipped_moab_version)
+  end
+
+  describe '#results' do
```

Comment:
I would expect to see somewhere (and maybe it's not here???) that the results are as expected for different cases:  when the result is at the endpoint;  when the result is not at the endpoint;   when the result is at the endpoint but doesn't match.

The naming of "services/audit/replication" has me expecting this to be here.

Reply:
All changes to the audit results are performed by `Audit::ReplicationSupport.check_child_zip_part_attributes` and `zip_endpoint.audit_class`. That these are invoked is tested here; the actions that they perform are not.